cmake/Utils.cmake

```diff
@@ -29,3 +29,52 @@ macro(add_target_source name)
     message(WARNING "Path ${CMAKE_CURRENT_LIST_DIR}/${name} does not exist")
   endif()
 endmacro()
+
+# Define a reusable macro for handling folder mappings
+macro(add_chimera_subdirectories target_platform category mappings)
+  # Initialize included folders
+  set(included_folders "")
+
+  # Process mappings
+  foreach(mapping IN LISTS ${mappings})
+    string(FIND "${mapping}" ":" delim_pos)
+    if(delim_pos EQUAL -1)
+      message(WARNING "[CHIMERA-SDK] Invalid mapping entry: '${mapping}'. Skipping.")
+      continue()
+    endif()
+
+
+    # Extract key and value
+    string(SUBSTRING "${mapping}" 0 ${delim_pos} key)
+    math(EXPR value_start "${delim_pos} + 1")
+    string(SUBSTRING "${mapping}" ${value_start} -1 value)
+
+    if(key STREQUAL "${target_platform}")
+      list(APPEND included_folders ${value})
+      break()
+    endif()
+  endforeach()
+
+  string(REPLACE "," ";" included_folders "${included_folders}")
+
+  # Align output with padding
+  string(LENGTH "[CHIMERA-SDK] Enabled ${category}s" category_prefix_length)
+  math(EXPR padding_length "36 - ${category_prefix_length}")
+  if(padding_length GREATER 0)
+    string(REPEAT " " ${padding_length} padding)
+  else()
+    set(padding "")
+  endif()
+
+  # Debug: Print the folders being included
+  message(STATUS "[CHIMERA-SDK] Enabled ${category}s${padding}: ${included_folders}")
+
+  # Add subdirectories, checking for a valid CMakeLists.txt
+  foreach(folder IN LISTS included_folders)
+    if(EXISTS ${CMAKE_CURRENT_LIST_DIR}/${folder}/CMakeLists.txt)
+      add_subdirectory(${folder})
+    else()
+      message(WARNING "[CHIMERA-SDK] ${category} folder '${folder}' does not contain a valid CMakeLists.txt. Skipping.")
+    endif()
+  endforeach()
+endmacro()
```

Comment:
This is clear against the declarative intent of the build system, but maybe I'm missing some insight... a switch case in the target is much easier to understand and debug, IMO, and requires no in-depth cmake knowledge.
Whether you keep this change, or revert to a simpler structure, please remove the commented version you end up not using.

Reply:
I argue that the macro makes it easier for non-experienced users to extend the SDK with new targets, as the data structure used in the end is much simpler and shorter (it is basically a dictionary in another language). This macro hides the complexity because CMake does not support dictionaries natively. But I also understand the downside and your concern. 

The problem I see is that we either have to do quite a lot of code duplication in the CMakeLists of the `drivers`, `devices` and `tests` folders to include the set of subfolders for each target or move it to a central point in the CMakeLists of the `targets` but this is not very intuitive as it is on a different folder hierarchy. Hence, I am in favor of the approach that is currently being implemented.

Ups, I forgot to delete the commented version in one place. This I can do for sure.

Reply:
I switched from a `macro` to a` function` to ensure that the parent scope is not unintentionally modified.